gradio/processing_utils.py

```diff
@@ -278,8 +277,16 @@ def save_file_to_cache(file_path: str | Path, cache_dir: str) -> str:
 def resolve_with_google_dns(hostname: str) -> str | None:
     url = f"https://dns.google/resolve?name={hostname}&type=A"
 
-    with urllib.request.urlopen(url) as response:
-        data = json.loads(response.read().decode())
+    if wasm_utils.IS_WASM:
+        import pyodide.http
+
+        content = pyodide.http.open_url(url)
+        data = json.load(content)
+    else:
+        import urllib.request
```

Comment:
I thought we used urllib originally because it was compatible with lite. Just wondering what the root cause is.

Reply:
It's confusing but network requests through `urllib` don't work on Pyodide by default, where `urllib3` works.